### ISSUE TITLE
Cluster Summarizer Workflow

### ISSUE THREAD

**Issue by dafajon** [GlobalMaksimum/sadedegel#29]
Cluster Summerizer is an unsupervised approach to extract informative sentences from text. In general the flow has two main levels:
* Generate sentence embeddings using pre-trained BERT model.
* Cluster the sentence embeddings as data points and pick the points closest to centers as informative sentences.

The levels are detailed as:

- [x]  Split the sentences from each other with sentece tokenizer. Form a `List[str]`.
- [x]  In each string add `[CLS]`(to beginning) and `[SEP]` (to end) tokens for BERT input format.
- [x]  Tokenize each string with `AutoTokenizer` from HuggingFace `transformers` and get a list of lists. `List[List[str]]`
- [x]  Convert `List[List[str]]` with `convert_tokens_to_ids` method to `List[List[int]]`
- [x]  Convert `List[List[int]]` to a `torch.Tensor` and pad up to maximum sentence length. Resulting tensor size: ` # of sentences in document x max_sent_len in document`. This finalizes input `token_ids`
- [x]   Prepare a tensor of same size where indexes that contain a token is 1 and a padding is 0. This finalizes input `token_type_ids.`
- [x]  `token_id`, `token_type_id` tensors are given to pre-trained BERT model where `output_hidden_states = True`.
- [ ]  12 Layers spit out embeddings of size 768 for each token in in each sentence. Resulting tensor size is `12 x # of sent in doc x max_sent_len x 768`
- [ ]  `select_layer` method selects which layer outputs to use.
- [ ]  Selected layer outputs and tokens are averaged to generate sentence embeddings. Resulting tensor size is `# of sent in doc x 768`.
- [ ]  `k = # of summary sentences` selected for KMeans clustering.
- [ ]  Embedding tensor is fed to clustering algorithm. 
- [ ]  Closest point to each center is detected.
- [ ]  Indices of the selected points are used to filter summary sentences.
